src/pykeen/models/unimodal/cp.py

```diff
@@ -0,0 +1,110 @@
+    citation:
+        author: Lacroix
+        year: 2018
+        arxiv: 1806.07297
+        link: https://arxiv.org/abs/1806.07297
+        github: facebookresearch/kbc
+    """
+
+    #: The default strategy for optimizing the model's hyper-parameters
+    hpo_default: ClassVar[Mapping[str, Any]] = dict(
+        embedding_dim=DEFAULT_EMBEDDING_HPO_EMBEDDING_DIM_RANGE,
+    )
+
+    def __init__(
+        self,
+        embedding_dim: int = 64,
+        rank: int = 64,
+        entity_initializer: Hint[Initializer] = None,
+        entity_initializer_kwargs: Optional[Mapping[str, Any]] = None,
+        entity_normalizer: Hint[Normalizer] = None,
+        entity_normalizer_kwargs: Optional[Mapping[str, Any]] = None,
+        relation_initializer: Hint[Initializer] = None,
+        relation_initializer_kwargs: Optional[Mapping[str, Any]] = None,
+        **kwargs,
+    ) -> None:
+        r"""Initialize CP via the :class:`pykeen.nn.modules.CPInteraction` interaction.
+
+        :param embedding_dim: The entity embedding dimension $d$.
+        :param rank: The tensor decomposition rank $k$.
+        :param entity_initializer: Entity initializer function. Defaults to None
+        :param entity_initializer_kwargs: Keyword arguments to be used when calling the entity initializer
+        :param entity_normalizer: Entity normalizer function. Defaults to None
+        :param entity_normalizer_kwargs: Keyword arguments to be used when calling the entity normalizer
+        :param relation_initializer: Relation initializer function. Defaults to None
+        :param relation_initializer_kwargs: Keyword arguments to be used when calling the relation initializer
+        :param kwargs: Remaining keyword arguments passed through to :class:`pykeen.models.ERModel`.
+        """
+        super().__init__(
+            interaction=CPInteraction,
+            entity_representations=[
+                # head representation
+                EmbeddingSpecification(
+                    shape=(rank, embedding_dim),
+                    initializer=entity_initializer,
+                    initializer_kwargs=entity_initializer_kwargs,
+                    normalizer=entity_normalizer,
+                    normalizer_kwargs=entity_normalizer_kwargs,
+                ),
+                # tail representation
+                EmbeddingSpecification(
+                    shape=(rank, embedding_dim),
+                    initializer=entity_initializer,
+                    initializer_kwargs=entity_initializer_kwargs,
+                    normalizer=entity_normalizer,
+                    normalizer_kwargs=entity_normalizer_kwargs,
+                ),
+            ],
+            relation_representations=EmbeddingSpecification(
+                shape=(rank, embedding_dim),
+                initializer=relation_initializer,
+                initializer_kwargs=relation_initializer_kwargs,
+            ),
+            skip_checks=True,
```

Comment:
would you please add a note why you're skipping checks (and a reminder what the checks are ;))

Reply:
> Since CP uses different representations for entities in head / tail role, the current solution is a bit hacky, and may be improved.

I am not sure whether we should keep this hack, or rather generalize the `ERModel` :thinking:

Reply:
For the scope of this PR let's keep the hack, but why don't you add a bit more information in a code comment with a  TODO so we could come back to this later

Reply:
Added note in 8308a58f